core/dictgen/src/rootcling_impl.cxx

```diff
@@ -4977,7 +4975,6 @@ int RootClingMain(int argc,
    {
       cling::Interpreter::PushTransactionRAII RAII(&interp);
       CI->getSema().getASTConsumer().HandleTranslationUnit(CI->getSema().getASTContext());
-      CI->clearOutputFiles(CI->getDiagnostics().hasErrorOccurred());
```

Comment:
Did you verify (i.e. try) that the PCM file gets removed?

Reply:
Likewise.